src/haddock/setup.py

```diff
@@ -0,0 +1,71 @@
+import toml
+import shutil
+import copy
+from pathlib import Path
+from haddock.error import SetupError
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Setup:
+    """Sets up the folder structure for the simulation."""
+
+    def __init__(self, workflow_f):
+        self.params = toml.load(workflow_f)
+
+    def validate(self):
+        # ====
+        # Major validation function
+        # ===
+
+        # check if steps are valid
+        module_directory = Path(__file__).parent / 'modules'
+        if 'order' not in self.params['input']:
+            raise SetupError("Workflow does not specify the"
+                             " order of execution")
+
+        steps = copy.deepcopy(self.params['input']['order'])
+        for step in steps:
+            module_loc = module_directory / step / 'default.py'
+            if not module_loc.exists():
+                logger.warning(f'Module {step} is not valid, it will'
```

Comment:
I am adding some comments here while I study and work on your PR.
I think *excluding* a module from the execution when it does not exist is **NOT** a good approach because this can facilitate passing silent errors because of typos or mistakes. In my opinion if a module specified by the user is not present we should **abort** the execution, and report the error to the user.

Reply:
Think so too, added this here just to develop things, will ammend it.